pagerduty/data_source_pagerduty_automation_actions_runner.go

```diff
@@ -0,0 +1,92 @@
+package pagerduty
+
+import (
+	"log"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourcePagerDutyAutomationActionsRunner() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourcePagerDutyAutomationActionsRunnerRead,
+
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"type": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"runner_type": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"creation_time": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"last_seen": {
+				Type:     schema.TypeString,
+				Computed: true,
+				Optional: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Computed: true,
+				Optional: true,
+			},
+			"runbook_base_uri": {
+				Type:     schema.TypeString,
+				Computed: true,
+				Optional: true,
+			},
```

Comment:
You should be able to reuse the resource schema directly here: https://github.com/PagerDuty/terraform-provider-pagerduty/pull/595/files#diff-c9f4179acaed6ea87351f857cb6180d4d178550a76e5e14bb54d18102123c33eR21-R65

And then reuse the `resourcePagerDutyAutomationActionsRunnerRead()` function too.

Reply:
For an example of how it is done in the Google terraform provider, see: https://github.com/hashicorp/terraform-provider-google/blob/v4.45.0/google/data_source_cloud_run_service.go

Reply:
While I agree this would be a less repetitive way of creating data source schemas I would vote that we keep the code base consistent with the other data sources in the PageDuty provider. So, if we were to adopt this approach we should use it for all the applicable data sources.